mergin/client.py

```diff
@@ -376,8 +384,23 @@ def download_project(self, project_path, directory):
         project_info = self.project_info(project_path)
         version = project_info['version'] if project_info['version'] else 'v0'
 
-        for file in project_info['files']:
-            self._download_file(project_path, version, file, directory)
+        # sending parallel requests is good for projects with a lot of small files
+        if parallel:
+            with concurrent.futures.ThreadPoolExecutor() as executor:
+                futures_map = {}
+                for file in project_info['files']:
+                    future = executor.submit(self._download_file, project_path, version, file, directory, parallel)
+                    futures_map[future] = file
+
+                for future in concurrent.futures.as_completed(futures_map):
+                    file = futures_map[future]
+                    try:
+                        future.result(60)
```

Comment:
How does this work? If a chunk fails to download within 60 seconds, the whole download will be cancelled? I am afraid that with slower connections this may be an issue...

Reply:
Indeed, it is mainly preparation for some retries on failures which we don't have yet. Shall we increase it or just leave it out at the moment?